run.sh

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
+
+# SIGTERM-handler
+term_handler() {
+  exit 0
+}
+trap term_handler SIGTERM
+
+if [ ! -f /sentinel/dash.conf ]; then
+  if [ -z "$RPCUSER" -o -z "$RPCPASSWORD" -o -z "$RPCPORT" ]; then
+    echo "When no /sentinel/dash.conf is present, you must at least set RPCUSER, RPCPORT and RPCPASSWORD environment variables"
+    exit 1
+  fi
+
+  echo "" > /sentinel/dash.conf
+  if [ -n "$RPCUSER" ]; then
+    echo "rpcuser=${RPCUSER}" >> /sentinel/dash.conf
+  fi
+  if [ -n "$RPCPASSWORD" ]; then
+    echo "rpcpassword=${RPCPASSWORD}" >> /sentinel/dash.conf
+  fi
+  if [ -n "$RPCPORT" ]; then
+    echo "rpcport=${RPCPORT}" >> /sentinel/dash.conf
+  fi
+fi
+
+if [ ! -f /sentinel/sentinel.conf ]; then
+  if [ -z "$RPCHOST" ]; then
+    echo "When no /sentinel/sentinal.conf is present, you must at least set the RPCHOST environment variable"
+    exit 1
+  fi
+
+  echo "dash_conf=/sentinel/dash.conf" > /sentinel/sentinel.conf
+  if [ -n "$RPCHOST" ]; then
+    echo "rpchost=${RPCHOST}" >> /sentinel/sentinel.conf
+  fi
+  if [ -n "$NETWORK" ]; then
+    echo "network=${NETWORK}" >> /sentinel/sentinel.conf
+  fi
+
+  # If these variables are not present, we let sentinel handle default values (which will use sqlite)
+  if [ -n "$DB_HOST" ]; then
+    echo "db_host=${DB_HOST}" >> /sentinel/sentinel.conf
+  fi
+  if [ -n "$DB_PORT" ]; then
+    echo "db_port=${DB_PORT}" >> /sentinel/sentinel.conf
+  fi
+  if [ -n "$DB_NAME" ]; then
+    echo "db_name=${DB_NAME}" >> /sentinel/sentinel.conf
+  fi
+  if [ -n "$DB_USER" ]; then
+    echo "db_user=${DB_USER}" >> /sentinel/sentinel.conf
+  fi
+  if [ -n "$DB_PASSWORD" ]; then
+    echo "db_password=${DB_PASSWORD}" >> /sentinel/sentinel.conf
+  fi
+  if [ -n "$DB_CHARSET" ]; then
+    echo "db_charset=${DB_CHARSET}" >> /sentinel/sentinel.conf
+  fi
+  if [ -n "$DB_DRIVER" ]; then
+    echo "db_driver=${DB_DRIVER}" >> /sentinel/sentinel.conf
+  fi
+fi
+
+while true; do
+  python ./bin/sentinel.py
+  sleep 5m
```

Comment:
5m should be fine, but technically the instructions advise running every minute because the built-in scheduler takes into account the necessary delay 'til next run.

Reply:
Also wondering if it makes sense to keep the loop/cron inside the Docker image vs calling externally for each run, and extracting the clock process.

Reply:
Changing it to 1m now. Can you explain more what you mean with "extracting the clock process."?

Reply:
I just mean doing a continuous docker run on the host machine via something like cron vs running it inside the container, so the Docker host would have something like `docker run -t --rm sentinel` instead of embedding the clock process (`run.sh`) in the image itself. Not sure which is ideal TBH, but your method would be simpler to just kick off Sentinel & forget about it.

Reply:
I like treating services in a way were they function on it's own as much as possible. Maybe that's even something for sentinel itself, so instead of calling it multiple times from the outside we change it to run like a daemon that has it's own service loop internally?

Reply:
In theory I agree. I think that I was trying to avoid the complexity of spawning a daemon, e.g. the spawning itself, signal handling, and mostly the process would be just sleeping.

So it was way easier to just call it from cron than to add the extra functionality, with minimal ROI/benefit. I generally lean toward a leaner code base if possible.